src/clusterfuzz/_internal/tests/core/bot/fuzzers/centipede/centipede_engine_test.py

```diff
@@ -99,16 +99,17 @@ class IntegrationTest(unittest.TestCase):
   """Integration tests."""
 
   def run(self, *args, **kwargs):
-    test_helpers.patch_environ(self)
     with get_test_paths() as test_paths:
       self.test_paths = test_paths
       os.environ['BUILD_DIR'] = str(self.test_paths.data)
       super().run(*args, **kwargs)
+      os.environ['BUILD_DIR'] = ''
```

Comment:
You could just move `os.environ['BUILD_DIR'] = str(self.test_paths.data)` into setUp and leave a comment as to why.

Reply:
I'm not sure that entirely works—the way the code is structured now, we're relying on that `with get_test_paths() as foo` bit to automagically destroy the temp directories as soon as it leaves that scope, so it's got to be called inside `run`, which means that variable needs to get inside `run`.

What *could* work, I think, is adding "destroy this tmppath" to tearDown; woudl that be preferable to you?

Reply:
I think moving os.environ['BUILD_DIR'] = str(self.test_paths.data) and patch_environ works. I just tested it.
I'm pretty sure this will get rid of the need to do `os.environ['BUILD_DIR'] = ''`
Otherwise, if you want to do something only involving setUp and tearDown and eliminating run entirely, that's fine with me.

Reply:
Huh, you're right.  I must've been confused about the ordering of run vs setup.  Fixed as asked!